src/api/spec/controllers/webui/cloud/azure/upload_jobs_controller_spec.rb

```diff
@@ -0,0 +1,230 @@
+        arch:            'x86_64',
+        filename:        'appliance.raw.xz',
+        target:          'azure',
+        image_name:      'image001',
+        subscription:    'myemailataws',
+        container:       'container001',
+        storage_account: 'mystorage',
+        resource_group:  'mygroup',
+        os_type:         'Linux'
+      }
+    end
+
+    shared_context 'it redirects and assigns flash error' do
+      before do
+        Feature.run_with_activated(:cloud_upload, :cloud_upload_azure) do
+          post :create, params: { cloud_backend_upload_job: params }
+        end
+      end
+
+      it { expect(flash[:error]).to match(/#{subject}/) }
+      it { expect(response).to be_redirect }
+    end
+
+    context 'without backend configured' do
+      subject { 'no cloud upload server configurated.' }
+
+      include_context 'it redirects and assigns flash error'
+    end
+
+    context 'with invalid parameters' do
+      context 'with an invalid filename' do
+        subject { 'apache2.rpm' }
+
+        before do
+          params[:filename] = subject
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid architecture' do
+        subject { 'i386' }
+
+        before do
+          params[:arch] = subject
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid image_name' do
+        subject { 'Image name not a valid format' }
+
+        before do
+          params[:image_name] = 'lorem ipsum'
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid subscription' do
+        subject { 'Subscription is too short' }
+
+        before do
+          params[:subscription] = 'no'
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid container' do
+        subject { 'Container not a valid format' }
+
+        before do
+          params[:container] = 'no valid container'
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid storage_account' do
+        subject { 'Storage account not a valid format' }
+
+        before do
+          params[:storage_account] = 'no valid storage account'
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid resource_group' do
+        subject { 'Resource group not a valid format' }
+
+        before do
+          params[:resource_group] = 'no valid resource group'
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+
+      context 'with an invalid os_type' do
+        subject { 'no valid OS type' }
+
+        before do
+          params[:os_type] = subject
+        end
+
+        include_context 'it redirects and assigns flash error'
+      end
+    end
```

Comment:
Above you are basically repeating the `Cloud::Azure::Params` spec as a controller spec. Why?

Reply:
Ask to @ChrisBr , I just mimic the same thing.

TBH, I think it will be needed for a later refactoring, I would like to convert all the UploadJob models into one and in this case the controller spec will become more important...

Reply:
That was 4 month ago ... I don't remember that! But yes, it looks like it is repeating the validation tests. I'm in favor of testing one parameter and dropping the others. I'm also fine with keeping it like it is.

> TBH, I think it will be needed for a later refactoring, I would like to convert all the UploadJob models into one and in this case the controller spec will become more important...

Why would you want to do that?

Reply:
>  I just mimic the same thing was done in EC2

Stop doing that. Do it *right[tm]* and fix up what is not right around your code. Or create an issue/card/FIXME/whatever for it.

Reply:
I thought it was good before and didn't check it really....

Reply:
I would test one parameter that it renders the flash.